ourhexenv.py

```diff
@@ -171,25 +172,31 @@ def reset(self):
             pygame.display.flip()
 
     def step(self, action):
-        #Check if the action is within the valid range.
+    # Check if the action is within the valid range
         if action not in range(self.action_space.n):
             raise ValueError("Illegal move: Action is out of bounds.")
-        # Handle pie rule
+
+        # Handle pie rule (only Player 2 can use it in the first round)
         if action == self.board_size * self.board_size:
             if self.agent_selection == "player_1":
                 raise ValueError("Illegal move: Pie rule can only be used by Player 2.")
             if not self.is_pie_rule_usable:
                 raise ValueError("Illegal move: Pie rule can only be used once.")
+            if np.sum(self.board == 1) == 0:
+                raise ValueError("Illegal move: Pie rule cannot be used before Player 1's move.")
```

Comment:
I don't think we are ever in this state. By definition of the environment, player 2 always goes after player 1. There's no way in which Player 2 is given a move before player 1